cmd/kubeadm/app/phases/addons/dns/dns_test.go

```diff
@@ -550,3 +553,230 @@ func createClientAndCoreDNSManifest(t *testing.T, corefile, coreDNSVersion strin
 	}
 	return client
 }
+
+func Test_deployedDNSReplicas(t *testing.T) {
+	type args struct {
+		dnsreplicas int32
+		replicas    int32
+	}
+	tests := []struct {
+		name           string
+		args           args
+		deploymentsize int
+		want           int32
+		wantErr        bool
+	}{
+		{
+			name: "one coredns addon deployment",
+			args: args{
+				dnsreplicas: 2,
+				replicas:    5,
+			},
+			deploymentsize: 1,
+			want:           2,
+			wantErr:        false,
+		},
+		{
+			name: "no coredns addon deployment",
+			args: args{
+				dnsreplicas: 2,
+				replicas:    5,
+			},
+			deploymentsize: 0,
+			want:           5,
+			wantErr:        false,
+		},
+		{
+			name: "multiple coredns addon deployments",
+			args: args{
+				dnsreplicas: 2,
+				replicas:    5,
+			},
+			deploymentsize: 3,
+			want:           5,
+			wantErr:        true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			client := createClient(t, tt.args.dnsreplicas, tt.deploymentsize)
+			got, err := deployedDNSReplicas(client, tt.args.replicas)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("deployedDNSReplicas() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if *got != tt.want {
+				t.Errorf("deployedDNSReplicas() = %v, want %v", *got, tt.want)
+			}
+		})
+	}
+}
+
+func createClient(t *testing.T, replisetsize int32, deploymentsize int) *clientsetfake.Clientset {
```

Comment:
```suggestion
// replicas is replica of each DNS deployment
// deploymentSize is the number of deployments with `k8s-app=kube-dns` label.
func newMockClientForTest(t *testing.T, replicas int32, deploymentSize int) *clientsetfake.Clientset {
```

To make the test func name more clear.